src/main/java/mate/academy/hibernate/relations/Main.java

```diff
@@ -15,18 +24,21 @@ public static void main(String[] args) {
         // use this session factory when you will initialize service instances
         SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
 
+        CountryDao countryDao = new CountryDaoImpl(sessionFactory);
         Country usa = new Country("USA");
-        CountryService countryService = null; // TODO: initialize this instance
+        CountryService countryService = new CountryServiceImpl(countryDao);
         countryService.add(usa);
```

Comment:
You don't need to save Dao layer to the variable in this case, so it could be inlined into (same for all DAOs in here):
 i
`CountryService countryService = new CountryServiceImpl(new CountryDaoImpl(sessionFactory));`